overlord/devicestate/devicemgr.go

```diff
@@ -608,12 +610,21 @@ func (m *DeviceManager) seedStart() (*timings.Timings, error) {
 
 func (m *DeviceManager) preloadGadget() (sysconfig.Device, *gadget.Info, error) {
 	var sysLabel string
-	modeEnv, err := maybeReadModeenv()
-	if err != nil {
-		return nil, nil, err
-	}
-	if modeEnv != nil {
-		sysLabel = modeEnv.RecoverySystem
+	if m.preseed {
+		var err error
+		sysLabel, err = systemForPreseeding()
```

Comment:
maybe we should stick something like preseedSystem on the manager and set it early so that we  don't have to call systemForPreseeding many times?